### PR TITLE
Sync configlet lint workflow

### PR DESCRIPTION
**Pull request by ErikSchierboom**
I've slightly updated the workflow to have, what I think, sensible names. This is what a workflow run now looks like: https://github.com/ErikSchierboom/fsharp/actions/runs/901814123

### CODE REVIEW

File: synced-files/.github/workflows/configlet.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Configlet
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  configlet:
+    name: Lint
```
Comment (ErikSchierboom) [inline synced-files/.github/workflows/configlet.yml:7]: I've added a name for the job, which was previously not there

File: synced-files/.github/workflows/configlet.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # 2.3.4
```
Comment (ErikSchierboom) [inline synced-files/.github/workflows/configlet.yml:12]: The previous version of the workflow did not have the `# 2.3.4` comment added.

**Comment by SaschaMann** [comment]
This needs to be moved to `tracks-files` if it's picked up again.

File: tracks-files/.github/workflows/configlet.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  configlet:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+
+    - name: Fetch configlet
+      uses: exercism/github-actions/configlet-ci@main
```
Comment (SaschaMann) [inline tracks-files/.github/workflows/configlet.yml:13]: Let's wait for https://github.com/exercism/configlet/pull/462

File: tracks-files/.github/workflows/configlet.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Configlet CI
+
+on: [push, pull_request, workflow_dispatch]
```
Comment (SaschaMann) [inline tracks-files/.github/workflows/configlet.yml:3]: I would prefer

```yaml
on:
  workflow_dispatch:
  push:
    branches:
      - main
  pull_request:
```

to not run builds that are useless at least 99.99999% of the time 🤺

Comment (ErikSchierboom) [reply tracks-files/.github/workflows/configlet.yml:3]: Agreed!

Comment (ErikSchierboom) [reply tracks-files/.github/workflows/configlet.yml]: Updated. I've also add the read permission

File: synced-files/.github/workflows/configlet.yml (hunk 2 above)
Comment (wolf99) [reply synced-files/.github/workflows/configlet.yml]: FYI that AFAIK Dependabot will not update version comments and they can then fall behind (quite quickly in some cases), which then turn them to bad information.

**Comment by ErikSchierboom** [comment]
I've updated this PR to use the reusable workflow that will be introduced in https://github.com/exercism/github-actions/pull/38

**Comment by SaschaMann** [comment]
For the `fmt` input to work, we'd have to make this file appendable.

**Comment by ErikSchierboom** [comment]
> For the fmt input to work, we'd have to make this file appendable.

Indeed. I'd be fine with doing that later though.

File: tracks-files/.github/workflows/configlet.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+name: Configlet CI
+
+on:
+  pull_request:
```
Comment (ErikSchierboom) [inline tracks-files/.github/workflows/configlet.yml:4]: @SaschaMann I've removed the branch restriction here too

**Review by SaschaMann — APPROVED** [review]
Since https://github.com/exercism/configlet/pull/462 is stale, let's go ahead and merge this already.

**Review by iHiD — APPROVED** [review]

**Review by iHiD — APPROVED** [review]

**Pull request merged**